src/utils/allocators.zig

```diff
@@ -587,16 +588,23 @@ pub const DiskMemoryAllocator = struct {
         // the allocator interface shouldn't allow this (aside from the *Raw methods).
         std.debug.assert(alignment <= std.mem.page_size);
 
-        const old_file_aligned_size = alignedFileSize(buf.len);
-        const new_file_aligned_size = alignedFileSize(new_size);
+        const old_file_aligned_size = alignedFileSize(buf.len + @sizeOf(Metadata));
```

Comment:
I'm a bit confused here, and in other places, this is done, `alignedFileSize` already adds `@sizeOf(Metadata)`, why are you doing it a second time?

Reply:
Negligence. I could have sworn this fixed an issue but I guess I was confused.

71f262b0a3922ba4a913a5313895789f97377c8d